docs/modules/operation/pages/deep-dive/topology/enlinkd/introduction.adoc

```diff
@@ -121,7 +137,15 @@ NOTE: If multiple protocols are enabled, the links will be discovered for each e
 3+| *Integer*
 
 | threads
-| Number of parallel threads used by updaters and link discovery.
+| Number of parallel threads used by schedulable: collector groups, updaters, and discovery.
```

Comment:
```suggestion
| Number of parallel threads used by schedulable: collector groups, updaters, and discovery.
```
Not sure what is meant by "schedulable". What is using the parallel threads? Suggest: "Number of parallel threads the schedule uses: collector, groups, updaters, and discovery." Or is it defined by the schedule?

Reply:
Well, 
We have three kinds of runnable in enlinkd, the first are TopologyUpdaters, they are scheduled using The LagacyScheduler every 5 minutes. Then we have BridgeTopologyDiscovery a runnable that is run to calculate the bbridge topology over Bridge data collected data. Finally we have SchedulableNodeCollectorGroup.java, well we define 6 SchedulableCollectionGroup one for every collection CDP, LLDP, BRIDGE, IPNETTOMEDIA, OSPF and IS-IS.
These classes are run every cdp_topology_interval etc etc, what do they do?

Each one generates a Specific Protocol Snmp Collection for every Node in opennms (that supports SNMP) assign a priority on the Snmp Collector and add this Collectors to the LegacyExecutors. The LegacyExecutors will run the collection using another thread pull and execution is done using the priority.

So we have two type of Obbject, the schedulable that are scheduled using the LegacyScheduler and the Executors (data collectors) that are executed/run) by another pull thread.

In fact in the confoguration file we have threads and executor-threads.

Reply:
Not sure if this answer your question. Just let me know, I leave this conversation as unresolved.

Reply:
Thank you! That is very helpful. I suggest the following:
```suggestion
| Number of parallel threads schedulable objects use: collector groups, updaters, and discovery.
```

Reply:
Many thanks Bonnie!